adjuster/src/main/java/org/jboss/pnc/reqour/adjust/App.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2024 Red Hat, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.jboss.pnc.reqour.adjust;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.picocli.runtime.annotations.TopCommand;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.jboss.pnc.api.enums.BuildType;
+import org.jboss.pnc.api.reqour.dto.AdjustRequest;
+import org.jboss.pnc.api.reqour.dto.AdjustResponse;
+import org.jboss.pnc.reqour.adjust.config.AdjustConfig;
+import org.jboss.pnc.reqour.adjust.config.ReqourAdjusterConfig;
+import org.jboss.pnc.reqour.adjust.provider.AdjustProvider;
+import org.jboss.pnc.reqour.adjust.provider.GradleProvider;
+import org.jboss.pnc.reqour.adjust.provider.MvnProvider;
+import org.jboss.pnc.reqour.adjust.provider.NpmProvider;
+import org.jboss.pnc.reqour.adjust.provider.SbtProvider;
+import org.jboss.pnc.reqour.common.utils.IOUtils;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+/**
+ * The entrypoint of the reqour adjuster.
+ */
+@TopCommand
+@CommandLine.Command(
+        name = "adjust",
+        description = "Execute the alignment with the corresponding built tool and manipulator",
+        mixinStandardHelpOptions = true,
+        versionProvider = VersionProvider.class)
+@Slf4j
+public class App implements Callable<AdjustResponse> {
+
+    @Inject
+    ReqourAdjusterConfig config;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @CommandLine.Parameters(description = "Adjust type", arity = "1")
+    BuildType type;
+
+    @CommandLine.Option(
+            names = { "-i", "--input-file" },
+            description = "Location of the input file (with the request). If not present, defaults to what is int the config (see 'reqour-adjuster.input-location').",
+            arity = "0..1")
+    Path inputLocation;
+
+    private static AdjustConfig adjustConfig;
+    private static AdjustRequest adjustRequest;
```

Comment:
Why not make these method paramaters?

Reply:
prefered this because I don't want this producer method

```java
    @Produces
    @ApplicationScoped
    AdjustProvider pickAdjustProvider() {
    ...
    }
```

to have any parameters and to use only class fields (since any parameter of a producer method is an injection point)